e2e/dractions/disapp.go

```diff
@@ -15,6 +15,14 @@ import (
 
 func EnableProtectionDiscoveredApps(ctx types.TestContext) error {
 	w := ctx.Workload()
+
+	d := ctx.Deployer()
+
+	disapp, ok := d.(*deployers.DiscoveredApp)
+	if !ok {
+		return fmt.Errorf("deployer is not a DiscoveredApp: %T", d)
+	}
```

Comment:
Please avoid this, we have IsDiscovered(), but we don't need to check the type here since the code calling should call us with the right deployer.